### PR TITLE
kubeadm addons: Bump coredns version

### PR DESCRIPTION
**Pull request by my-git9**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

CoreDns 1.11.4 released: https://github.com/coredns/coredns/releases/tag/v1.11.4

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Comment by kl52752** [comment]
Test is failing with image pull error
```
I1113 05:59:28.390106 65658 dump.go:53] At 2024-11-13 05:48:21 +0000 UTC - event for coredns-74bc44c9bd-lvxml: {kubelet e2e-0db0337a64-674b9-minion-group-21zp} Failed: Error: ErrImagePull
  I1113 05:59:28.390117 65658 dump.go:53] At 2024-11-13 05:48:21 +0000 UTC - event for coredns-74bc44c9bd-lvxml: {kubelet e2e-0db0337a64-674b9-minion-group-21zp} Pulling: Pulling image "registry.k8s.io/coredns/coredns:v1.11.4"
  I1113 05:59:28.390130 65658 dump.go:53] At 2024-11-13 05:48:21 +0000 UTC - event for coredns-74bc44c9bd-lvxml: {kubelet e2e-0db0337a64-674b9-minion-group-21zp} Failed: Failed to pull image "registry.k8s.io/coredns/coredns:v1.11.4": rpc error: code = NotFound desc = failed to pull and unpack image "registry.k8s.io/coredns/coredns:v1.11.4": failed to resolve reference "registry.k8s.io/coredns/coredns:v1.11.4": registry.k8s.io/coredns/coredns:v1.11.4: not found
  ```

**Comment by my-git9** [comment]
> Test is failing with image pull error
> 
> ```
> I1113 05:59:28.390106 65658 dump.go:53] At 2024-11-13 05:48:21 +0000 UTC - event for coredns-74bc44c9bd-lvxml: {kubelet e2e-0db0337a64-674b9-minion-group-21zp} Failed: Error: ErrImagePull
>   I1113 05:59:28.390117 65658 dump.go:53] At 2024-11-13 05:48:21 +0000 UTC - event for coredns-74bc44c9bd-lvxml: {kubelet e2e-0db0337a64-674b9-minion-group-21zp} Pulling: Pulling image "registry.k8s.io/coredns/coredns:v1.11.4"
>   I1113 05:59:28.390130 65658 dump.go:53] At 2024-11-13 05:48:21 +0000 UTC - event for coredns-74bc44c9bd-lvxml: {kubelet e2e-0db0337a64-674b9-minion-group-21zp} Failed: Failed to pull image "registry.k8s.io/coredns/coredns:v1.11.4": rpc error: code = NotFound desc = failed to pull and unpack image "registry.k8s.io/coredns/coredns:v1.11.4": failed to resolve reference "registry.k8s.io/coredns/coredns:v1.11.4": registry.k8s.io/coredns/coredns:v1.11.4: not found
> ```

The coredns:v1.11.4 image should have been uploaded, thanks

**Comment by aojea** [comment]
the image was promoted recently https://github.com/kubernetes/k8s.io/pull/7519

File: cmd/kubeadm/app/phases/addons/dns/dns_test.go [hunk 1]
```diff
@@ -1423,28 +1423,28 @@ func TestDeployedDNSAddon(t *testing.T) {
...
-			image:          "registry.k8s.io/coredns/coredns:v1.11.3",
+			image:          "registry.k8s.io/coredns/coredns:v1.11.4",
 			deploymentSize: 2,
 			wantVersion:    "",
 			wantErr:        true,
 		},
 		{
 			name:           "with digest",
-			image:          "registry.k8s.io/coredns/coredns:v1.11.3@sha256:a0ead06651cf580044aeb0a0feba63591858fb2e43ade8c9dea45a6a89ae7e5e",
+			image:          "registry.k8s.io/coredns/coredns:v1.11.4@sha256:a0ead06651cf580044aeb0a0feba63591858fb2e43ade8c9dea45a6a89ae7e5e",
```
Comment (DamianSawicki) [inline cmd/kubeadm/app/phases/addons/dns/dns_test.go:1445]: With the new patch version, sha256 should also be updated.

**Review by DamianSawicki — CHANGES_REQUESTED** [review]

**Comment by my-git9** [comment]
dup:https://github.com/kubernetes/kubernetes/pull/128926

**Pull request closed without merging**